### ISSUE TITLE
Indices Exists API missing include_type_name in REST API spec

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch#43905]
### Issue

Relates: https://github.com/elastic/elasticsearch-net/issues/3901

The REST API spec for indices.exists.json is missing `include_type_name` in 6.8.0:

https://github.com/elastic/elasticsearch/blob/12632795cdeb57a9092da4a0508f58424c5d7fc6/rest-api-spec/src/main/resources/rest-api-spec/api/indices.exists.json#L16-L42

Sending an index exists request without `include_type_name` results in a Deprecation Warning HTTP header being returned:

Request

```
HEAD http://127.0.0.1:9200/foo HTTP/1.1
Accept: application/json
Content-Type: application/json
Host: 127.0.0.1:9200
Connection: Keep-Alive
```

Response

```
HTTP/1.1 200 OK
Warning: 299 Elasticsearch-6.8.1-1fad4e1 "[types removal] The parameter include_type_name should be explicitly specified in get indices requests to prepare for 7.0. In 7.0 include_type_name will default to 'false', which means responses will omit the type name in mapping definitions."
content-type: application/json; charset=UTF-8
content-length: 225
```

Sending a request with `include_type_name=true` or `include_type_name=false` results in no Deprecation Warning.

### Feature

The indices.exists.json REST API spec should include `include_type_name` in the list of params, so that clients that generate API methods from the spec pick this up.

**Comment by cbuescher** [comment]
@russcam instead of adding the parameter to the indices.exist API, we decided to rather prevent the deprecation log for the exists API (see discussion in #43910 and fix in #43963). Just backporting this to 6.8 now, if we do another minor release on that branch that might help the user. It doesn't really make sense to have this parameter in this API since the body is empty. Let me know if this you think this solved the issue or if we are missing something.

**Comment by russcam** [comment]
Removing the deprecation warning for index.exists is a much better solution. Thanks for the fix @cbuescher  👍
